framework/src/utils/ADFParser.C

```diff
@@ -0,0 +1,36 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADFParser.h"
+
+ADFParser::ADFParser() : FunctionParserAD(), _epsilon(1e-12) {}
+
+ADFParser::ADFParser(const ADFParser & cpy) : FunctionParserAD(cpy), _epsilon(1e-12) {}
+
+bool
+ADFParser::JITCompile()
+{
+  auto result =
+      JITCompileHelper("DualReal",
+                       ADFPARSER_INCLUDES,
+                       "#include \"DualReal.h\"\n#include \"metaphysicl/dualnumberarray.h\"\n");
```

Comment:
@dschwen I should have caught this on review. Just `#include ADReal.h` and that should get you everything you need.

Reply:
ah, cool, I'll make a fixup PR